plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/SessionPropertyConfigurator.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hudi.testing;
+
+import io.trino.Session;
+
+import static io.trino.plugin.hudi.HudiSessionProperties.COLUMN_STATS_INDEX_ENABLED;
+import static io.trino.plugin.hudi.HudiSessionProperties.DYNAMIC_FILTERING_WAIT_TIMEOUT;
+import static io.trino.plugin.hudi.HudiSessionProperties.METADATA_TABLE_ENABLED;
+import static io.trino.plugin.hudi.HudiSessionProperties.PARTITION_STATS_INDEX_ENABLED;
+import static io.trino.plugin.hudi.HudiSessionProperties.QUERY_PARTITION_FILTER_REQUIRED;
+import static io.trino.plugin.hudi.HudiSessionProperties.RECORD_LEVEL_INDEX_ENABLED;
+import static io.trino.plugin.hudi.HudiSessionProperties.SECONDARY_INDEX_ENABLED;
+import static java.util.Objects.requireNonNull;
+
+public class SessionPropertyConfigurator
```

Comment:
nit: rename to `SessionFactory` or `SessionProvider`?

Reply:
Renaming it to `SessionBuilder` as this somewhat follows a builder pattern